File: 03-Append Characters to String to Make Subsequence/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int appendCharacters(string s, string t) {
        int s_pointer = 0, t_pointer = 0;

        while (s_pointer < s.length() && t_pointer < t.length()) {
            if (s[s_pointer] == t[t_pointer]) t_pointer++;
            s_pointer++;
        }

        return t.length() - t_pointer;
    }
};

File: 03-Append Characters to String to Make Subsequence/solution.py
class Solution:
    def appendCharacters(self, s, t):
        s_pointer = 0
        t_pointer = 0

        while s_pointer < len(s) and t_pointer < len(t):
            if s[s_pointer] == t[t_pointer]:
                t_pointer += 1
            
            s_pointer += 1

        return len(t) - t_pointer